src/bitcoin-tx.cpp

```diff
@@ -395,27 +395,42 @@ static void MutateTxAddOutMultiSig(CMutableTransaction& tx, const std::string& s
 static void MutateTxAddOutData(CMutableTransaction& tx, const std::string& strInput)
 {
     CAmount value = 0;
+ 
+    // separate [VALUE:]DATA[:ASSET] in string
+    std::vector<std::string> vStrInputParts;
+    boost::split(vStrInputParts, strInput, boost::is_any_of(":"));
 
-    // separate [VALUE:]DATA in string
-    size_t pos = strInput.find(':');
-
-    if (pos==0)
+    // Check that there are enough parameters
+    if (vStrInputParts.size()<1)
```

Comment:
nit: just check `vStrInputParts.empty()`

Reply:
Is it even possible that that is empty? I don't know about boost specifically, but most string splitters do `"".split(":") == [""]` so it would always have 1 element, even for an empty string.